my-diploma/src/customHooks/useFetching.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useCallback } from "react";
+
+export const useFetching = (api, formValue, resetForm) => {
```

Comment:
1. this hook not only for forms. so that use appropriate naming for your params.
2. api here is not a noun, it is a method, use appropriate naming for functions based on js naming rules